edwards25519.go

```diff
@@ -121,14 +121,21 @@ func (v *Point) Set(u *Point) *Point {
 // Bytes returns the canonical 32 bytes encoding of v, according to RFC 8032,
 // Section 5.1.2.
 func (v *Point) Bytes() []byte {
+	// This function is outlined to make the allocations inline in the caller
+	// rather than happen on the heap.
+	var out [32]byte
+	return v.bytes(&out)
+}
+
+func (v *Point) bytes(out *[32]byte) []byte {
 	var recip, x, y fieldElement
 	recip.Invert(&v.z)
 	x.Multiply(&v.x, &recip) // x = X / Z
 	y.Multiply(&v.y, &recip) // y = Y / Z
 
-	out := y.Bytes()
+	y.bytes(out)
```

Comment:
Rename `out` to `buf` here and in the caller, and then make `out` the returned slice, so you can just `return out` below.

```suggestion
	out := y.bytes(buf)
```

Reply:
done